### DISCUSSION TITLE
Parse CCD document using Mirth Connect 3.8.1.

### DISCUSSION THREAD

**Discussion by smita-srivastava** [nextgenhealthcare/connect#4627] [Q&A]
Need to parse the CCD document using Mirth Connect 3.8.1. I'm trying to use MDHT but not able to find enough help online. Can anyone provide reference/samples for working with CCDA documents in MIRTH community version?

**Comment by rjkroll** [comment]
What have you already tried?

You essentially need to do the following things:

Have you installed the MDHT Run-time JARS custom-lib folder?

How are your channel datatypes setup?  
  - They should be HL7 v3.0 with Strip Namespaces unchecked

How are you initializing the MDHT CCD package in your Deploy script?
 // This script executes once when the channel is deployed
// You only have access to the globalMap and globalChannelMap here to persist data

// Initialize MDHT Clinical Document Package
org.openhealthtools.mdht.uml.cda.ccd.CCDPackage.eINSTANCE.eClass();

return;

How is your Source transformer loading the document and parsing the section/fields to variables? 

// Load the CCD Document
var doc = org.openhealthtools.mdht.uml.cda.util.CDAUtil.load(new java.io.ByteArrayInputStream(messageObject.getRawData().getBytes("UTF-8")));

// Get CCD Document Sections to be parsed
var docPatientRole = doc.getRecordTargets().get(0).getPatientRole();
var docPatient = docPatientRole.getPatient();
var docPatientName = docPatient.getNames().get(0);

// Map Patient Identity Fields to Mirth Channel Map Variables
channelMap.put('patientFirstName', docPatientName.getGivens().get(0).getText());
channelMap.put('patientLastName', docPatientName.getFamilies().get(0).getText());
channelMap.put('patientGenderCode', docPatient.getAdministrativeGenderCode().getCode());
channelMap.put('patientDateOfBirth', docPatient.getBirthTime().getValue()); // YYYYMMDD

**Reply by pacmano1** [reply]
@smita-srivastava  I would add that the etiquette of the forum applies here also, see https://forums.mirthproject.io/articles/177082-forum-etiquette

In particular bullets 8,9,10,11.

**Reply by smita-srivastava** [reply]
Point noted @pacmano1. This was my first post in this forum and I was not aware of the etiquette. Thank you for letting me know.

@rjkroll Thanks for your response. I already tried all the steps mentioned at https://www.spheregen.com/parsing-continuity-of-care-ccd-using-mirth-and-mdht/ but it was giving error. 

I was able to fix the issue by replacing messageObject with connectorMessage while declaring doc object:

var doc = org.openhealthtools.mdht.uml.cda.util.CDAUtil.load(new java.io.ByteArrayInputStream(connectorMessage.getRawData().getBytes("UTF-8")));

Thanks.
